src/Core/Checkout/Customer/SalesChannel/SwitchDefaultAddressRoute.php

```diff
@@ -97,13 +106,19 @@ public function swap(string $addressId, string $type, SalesChannelContext $conte
                     'defaultBillingAddressId' => $addressId,
                 ];
 
+                $event = new CustomerSetDefaultBillingAddressEvent($context, $customer);
+                $this->eventDispatcher->dispatch($event);
+
                 break;
             default:
                 $data = [
                     'id' => $customer->getId(),
                     'defaultShippingAddressId' => $addressId,
                 ];
 
+                $event = new CustomerSetDefaultShippingAddressEvent($context, $customer);
```

Comment:
same here

Reply:
Also here the customer entity object is itself passed